File: frontend/src/types/response.ts
import type { Appliance, Cookware, Utensil } from "./request";

export type RecipeStep = {
  number: number;
  details: string;
};

export type Recipe = {
  name: string;
  description: string;
  appliances: Appliance[];
  cookware: Cookware[];
  utensils: Utensil[];
  ingredients: string[];
  steps: RecipeStep[];
};

export type Response = {
  recipes: Recipe[];
};

File: backend/Models/Response.cs
namespace backend.Models;

public record RecipeStep(int Number, string Details);

public record Recipe(
    string Name,
    string Description,
    List<string> Appliances,
    List<string> Cookware,
    List<string> Utensils,
    List<string> Ingredients,
    List<RecipeStep> Steps
);

public record GenerateRecipeResponse(List<Recipe> Recipes);
